components/apim-analytics/org.wso2.analytics.solutions.apim.analytics/widgets/APIMApiBackendUsageSummary/src/CustomTable.jsx

```diff
@@ -0,0 +1,288 @@
+    },
+    table: {
+        minWidth: 200,
+    },
+    tableWrapper: {
+        overflowX: 'auto',
+    },
+    loadingIcon: {
+        margin: 'auto',
+        display: 'block',
+    },
+    paginationRoot: {
+        color: theme.palette.text.secondary,
+        fontSize: theme.typography.pxToRem(12),
+        '&:last-child': {
+            padding: 0,
+        },
+    },
+    paginationToolbar: {
+        height: 56,
+        minHeight: 56,
+        padding: '0 5%',
+    },
+    paginationCaption: {
+        flexShrink: 0,
+    },
+    paginationSelectRoot: {
+        marginRight: '10px',
+    },
+    paginationSelect: {
+        paddingLeft: 8,
+        paddingRight: 16,
+    },
+    paginationSelectIcon: {
+        top: 1,
+    },
+    paginationInput: {
+        color: 'inherit',
+        fontSize: 'inherit',
+        flexShrink: 0,
+    },
+    paginationMenuItem: {
+        backgroundColor: theme.palette.type === 'light' ? '#fff' : '#162638',
+    },
+    paginationActions: {
+        marginLeft: 0,
+    },
+});
+
+/**
+ * Create React Component for Api Backend Usage Summary Table
+ */
+class CustomTable extends React.Component {
+    /**
+     * Creates an instance of CustomTable.
+     * @param {any} props @inheritDoc
+     * @memberof CustomTable
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            tableData: [],
+            page: 0,
+            rowsPerPage: 5,
+            orderBy: 'hits',
+            order: 'desc',
+            expanded: false,
+            filterColumn: 'apiname',
+            query: '',
+        };
+    }
+
+    handleRequestSort = (event, property) => {
+        const { order, orderBy } = this.state;
+        let orderNew = 'desc';
+        if (orderBy === property && order === 'desc') {
+            orderNew = 'asc';
+        }
+        this.setState({ order: orderNew, orderBy: property });
+    };
+
+    handleChangePage = (event, page) => {
+        this.setState({ page });
+    };
+
+    handleChangeRowsPerPage = (event) => {
+        this.setState({ rowsPerPage: event.target.value });
+    };
+
+    handleExpandClick = () => {
+        this.setState(state => ({ expanded: !state.expanded }));
+    };
+
+    handleColumnSelect = (event) => {
+        this.setState({ filterColumn: event.target.value });
+    };
+
+    handleQueryChange = (event) => {
+        this.setState({ query: event.target.value });
+    };
+
+    /**
+     * Render the Api Backend Usage Summary table
+     * @return {ReactElement} customTable
+     */
+    render() {
+        const { data, classes } = this.props;
+        const {
+            query, expanded, filterColumn, order, orderBy, rowsPerPage, page,
+        } = this.state;
+
+        const lowerCaseQuery = query.toLowerCase();
+
+        this.state.tableData = query
+            ? data.filter(x => x[filterColumn].toString().toLowerCase().includes(lowerCaseQuery))
+            : data;
+        const { tableData } = this.state;
```

Comment:
this can be moved to L188

Reply:
In line 193, ``tableData`` is setted. Hence we need that new value. So it is included in line 196.